Ancestral sample

DL42 opened the pull request:
reorganized ancestral sample branch

molpopgen committed:
Merge pull request #18 from DL42/ancestral_sample

Ancestral sample

DL42 committed:
Merge branch 'dev' into ancestral_sample

DL42 committed:
starting c++ ancestral sampling, adding mutation struct

Commit:
adding mutations to ancestry_tracker.hpp

Commit:
adding mutations to wfarg, changing description of EdgesetTable to EdgeTable

DL42 committed:
gitignore

Commit:
changed mutation structure to store node_id and the invariant position of the mutation in fwdpp vector (which in turn has the position and metadata info)

Commit:
adding new handler for combined recombination/mutation, remove old handler

Commit:
changing from recmodel to generate_breakpoints

Commit:
adding mutations via generate_new_mutations rather than mutate_gamete_recycle

Commit:
adding mutations to handle_mut_rec

DL42 committed:
add KTfwd namespace to generate_new_mutations and breakpoints function

also reorder lines/spaces for clarity

DL42 committed:
change recombine_gametes to mutate_recombine

DL42 committed:
removal of update_indexes

shouldn't be necessary if arg_tracker is constructed with correct starting index (only needed for starting simulation with initialized tree as opposed to a blank simulation)

DL42 committed:
clear mutations after gc

also fixes a couple of code comments

DL42 committed:
adding mutation tables to argsimplifier

adding mutation and site tables using info from ancestry tracker and optional initial tree with metadata for each

metadata for ancestry tracker is pickled fwdpy11 mutation type, default metadata for mutations from initial tree is added if none present (as would be if tree from msprime)

Commit:
remove multithreaded versions (for now)

code in wfarg and setup commented out, removed in evolve_arg, README updated

Commit:
modified call to accept pop instead of generation

Commit:
move final simplification to inside wright_fisher_single_thread

provide an override variable to __call__ which overrides the gc interval and forces simplification

Commit:
bug fixes to let code run (no mutations present - current setting in test_evolve)

edgesets to edges in test_evolve, import namedtuple, check to make sure mutation array has mutations, parantheses errors,
small change to README, gitignore ignores pyc files, comment out async/queue functions in evolve_functions header, slight addendum to override parameter

Commit:
program run when testing w/ mutations

Commit:
simplify sites and mutations for final sample

molpopgen committed:
call recombination model directly, which avoids a fwdpp-based idiosyncrasy

molpopgen committed:
import pickle and fix call to get numpy rep of pop's mutationsm

DL42 committed:
Merge pull request #1 from ThorntonLab/ancestral_sample

Make sure breakpoints are generated

DL42 committed:
added comment describing pickling pattern of mutations

also added caculating time_appending back in

DL42 committed:
amended comment describing pickling of mutations

changes in fwdpy11 allow easy construction of fwdpy11 mutation type from unpickled metadata

Commit:
simulate with initial_tree (for testing purposes)

Commit:
subsume ancestry_tracker inside wright-fisher
use ancestry_processor's num_nodes to set next_index (if next_index == 0, initialize a tracker with newly created "parent" nodes)

Commit:
return to default of not simulating with initial tree

Commit:
fixed bug introduced in commit 309f1096c7aa27677483142687ed402f7b1b17a4
when not initializing tree need to set next_index to 2N after initializing tracker with 2N nodes
minor: additional check in argsimplifier initializer when adding default metadata that there are mutations (generally not a problem before if there weren't, simply created an empty list, but might as well not create anything if uneeded)

Commit:
changed ancestry_tracker to store nodes in reverse time, so need to reverse time or increment it during simplification
increment time of nodes inherited from any previous simulation during argsimplifier initialization

Commit:
moves garbage collection to the end of the generation

(with the previous change in setting reverse time right off the bat, gcing at the beginning vs end makes no difference, but this way if gc is a multiple of the total number of generations, you don't have to do a final simplification step)

Commit:
comment out exchange_for_async and ancestry_data (last of the async/queue items left)

Commit:
removing extra last_gc_time and process variables from argsimplifier and ancestry tracker

Commit:
simplified node API and usage

Commit:
simplify mutation API

Commit:
simplify edge API and fixes no recombination bug

previously if there were no breakpoints, mutations would not be added properly to the gamete or the ancestry tracker in handle_mut_rec, this change fixes that

Commit:
removed lastN from ancestry_tracker

didn't seem to do anything (function can be replaced by upcoming generation & pop indexes matrix if it becomes needed)

Commit:
adding a vector of vector of ints (gen_pop_indexes) to store each generation's next ID

replaces lastN and first_parental_index, will replace offspring_indexes
will be used to grab ancestral samples
when bookkeeping for multiple populations, access will be strided - e.g. p1g1, p2g1, ..., p1g2, p2g2, etc ...
also: fixed variable __gc_interval so that it corresponds to the same variable in the getter/setter and style of the other argsimplifier variables

Commit:
replace offspring indexes with pop_gen_indexes

samples are created by the python range function between [ pop_gen_indexes[index_gen-1], pop_gen_indexes[index_gen] )
ancestral samples will be taken from:
displacement = generation - ancestral_sample_gen
random sample from interval [ pop_gen_indexes[index_gen- displacement - 1], pop_gen_indexes[index_gen - displacement] )

Commit:
using pair instead of array to store node indexes for a generation

We will sample ancestral nodes on the C++-side and store the sample node indexes in ancestry tracker in an array. Therefore, for the current generation's node IDs, ancestry tracker only needs to store the start-end node IDs for a given generation rather than the previous per-population, per-generation node index system.

Commit:
testing ancestral samples (doesn't work)

Commit:
reorganizing python code around argsimplifier so that simulations are run through the argsimplifier

streamlines user-facing API while still letting fwdpy11 and ancestry_tracker objects be declared in python scope
small modifications to C++ code to reflect changed python organization

Commit:
renamed argsimplifier argevolver to reflect changes in API organization

C++ evolve_singlepop_regions_track_ancestry still calls it simplifier as the function is passed the simplifier call function (its main purpose in that simulation)

Commit:
adding a python-based gamete sampler to argevolver for taking ancestral samples

Commit:
created sampler that sample X individuals every Y generations (current example is 4 individuals every 125 generations)

sample at end of the generation as opposed to the beginning
don't need any extra bookkeeping in ancestry_tracker or evolve_singlepop_regions_track_ancestry
moved post_process_gc to the end of the simplification
performing an ancestral sample forces a GC that generation

Commit:
added separate seed for sampler

made final sample seed, seed+1
should make the final sample invariant to whether or not ancestral samples are taken

Commit:
create and draw trees, check for mutation duplicates

Commit:
fix mutation duplicate issue by independently storing mutation position in the ancestry track's mutation struct and using that rather than the position from the population mutation vector to set the needed information in msprime's MutationTable

Commit:
updating simple prototype with mutations to test msprime & mutations

Commit:
preprocess mutation vector to remove mutations from the ancestry tracker that have already been removed from the pop mutation vector

vastly improves append time, minor improvement to simplify and sort, negligible increase in prep time

Commit:
allows ancestral samples to be taken of nodes from input trees

Commit:
empty brace initialization for nodes, edges, mutations in ancestry_tracker
print evolver.sites.num_rows, quick check to make sure simulation is proceeding okay
comment out duplicate checker for now
use home to set draw path

Commit:
add function to preserve mutation indices of ancient samples created during the simulation

Commit:
custom make_mut_queue: add mutations to the mutation queue if they have 0 count *AND* if they are not in the preservation lookup

Commit:
custom update_mutations: remove mutation positions from lookup if have 0 count *AND* if not in preservation lookup

Commit:
Revert "allows ancestral samples to be taken of nodes from input trees"

This reverts commit 68eaf7824e1e2c87c5c1c05f150a3b3ca4ef16a8.
Commit doesn't work for sampling from input trees, current sampler is meant to be used as a sampler for individuals, not nodes (gametes) -> input trees won't necessarily have gametes of individuals consecutive in the NodeTable
If user wants to ensure the root node is present in final tree (good for tracking neutral fixations), user can set the ancestral sample for generation 0 to take all pop.N  individuals as samples

Commit:
make total_generations an attribute of argevolver, use as one point of reference for total_generations in rest of simulation

Commit:
rearranges _anc_sampler to be a member of argevolver's __init and __call__ functions
removes need to GC every time an ancestral sample is taken, so GC is just a function of GC interval
in __call__ remove override and check if pop.generation is at final generation, then simplify
removes need to check if length of new  edges or nodes > 0 (only possible if total population in simulation is 0 between two GCs, which is a malformed simulation and should be checked for and error thrown if happens)
gives option of taking a final sample of the population (by taking a sample at total_generations) or passing on the full final generation (by not taking a sample at total_generations) ... would not recommend taking a sample at total_generations AND then tried to take another subsample of that sample afterwards (especially in the case of taking ancestral samples), likely result in msprime error of duplicate leaf/sample nodes

Commit:
return pop from ArgEvolver

Commit:
only encode mutation_id into metadata for MutationTable

no need to encode full mutation information if mutation has been preserved in pop.mutations, just record its index! (cuts appending time in half)
(may be even faster to not use pickle, encode the indices and offsets directly ...)

Commit:
prototype merge_sites function

Commit:
preserve fixations

Commit:
replaces pickle with struct, packs the pos.mutation index into an int formatted byte string
test_evolve counts duplicates and sites without matching metadata
prints first selected tree with mutations and the corresponding tree in the neutral set of trees

Commit:
remove copy of pop.mutation.array as it is no longer needed during simplification

Commit:
take in reference to pybind11 array in preserve_mutations_sample

Commit:
faster metadata creation using numpy.view (still uses struct.unpack for getting back the mutation_id)

performance of simulation with metadata is now practically no different than performance without storing metadata

Commit:
update to fwdpy11 1.5: compiles, no idea if it runs

Commit:
removed deleted files from setup.py

Commit:
simulation runs

Commit:
replaced fwdpp's infsites_popgenmut with infsites_Mutation from fwdpy11

Commit:
upgraded update_mutations to match similar function in latest fwdpy11

Commit:
adding extra diploid fields

Commit:
changed fitness function to multiplicative since we are removing fixations from gametes and set fitness using diploid's w field in evolve_generation's main loop instead of calculating it in w

Commit:
preserve all fixations and test lookup table

Commit:
mcounts will be set to 0 next generation anyway as gamete_cleaner removes fixations from gametes ... and I'm preserve fixations in the mutation array anyway since they'll be in the MutationTable

Commit:
removed mutation metadata ... with the numpy view trick, storing/accessing single integer metadata was fast now, but if there is no reason to have it with the lookup table ... then might as well remove it

Commit:
initial injection of juvenile migration code with two demes, seems to work

Commit:
small two-population simulation with human out-of-africa demography, segmentation fault 11

(burn-in time and 2nd population/migration times reduced by factor of 10)

Commit:
default simulation runs

(burn-in time and 2nd population/migration times reduced by factor of 10)
append taking up 5% of runtime
{'prepping': 0.17922800000002914, 'sorting': 36.00791799999996, 'appending': 5.656881000000087, 'simplifying': 19.648757999999958, 'simulating': 42.547677000000256}

Commit:
expand final population size

Commit:
set migration boundaries to prevent too large/small migration

Commit:
small changes

Commit:
alterations to reflect underlying changes in fwdpy11 population and parameter classes

Commit:
update reflects changes in fwdpy11 diploid class (separating metadata from gametes)

Commit:
fwdpy11 changed parental data tuple to parent array

Commit:
error in previous commit

Commit:
np.random.choice gave error on linux dev machine when final_pop2_size was 0

Commit:
added extra print statement to printout number of rows, fixed evolver.pop.mut_lookup (is just a dictionary, not function?)

Commit:
requirements file

Commit:
more flexible initial starting points

Commit:
fixing string to int bugs

Commit:
added neutral theta as parameter

Commit:
bug fix

Commit:
bug fix2

Commit:
fixed input bug for pop1 and when making a single population simulation

DL42 committed:
initial upgrades to create parallel simulations

DL42 committed:
fixes + make burn-in a parameter

DL42 committed:
tab error will do full fix later

DL42 committed:
more tab error

DL42 committed:
name error

DL42 committed:
fixing seeds

DL42 committed:
putting back in accidently removed lines

DL42 committed:
fix seeds and add print

DL42 committed:
hopefully a fix to the numpy dtype error

DL42 committed:
should fix dtype error

DL42 committed:
absolutely should fix dtype error

DL42 committed:
name error

Commit:
test evolver in interactive window

Commit:
made sure anc_sampling was of type int

Commit:
fixing tabs

Commit:
fixing parsed args to args

Commit:
moved anc_sam conversion to int to evolve_arg and fixed indexing in sample_index

Commit:
fix sample taking

Commit:
fix sample dtype

Commit:
create sample list from node list

Commit:
change results to evolver

DL42 committed:
first commit for multi-sample fst analysis

Commit:
syntax and bug fixes

Commit:
(test_evolve and evolve_arg) moved args.anc_sam1/2 manipulation back to test_evolve, add burn_in generations to the args.anc_sam1/2
(argevolver) for max sample index add pop_size1 and 2 instead of just using __pop.N

Commit:
eliminated last reference to burn-in scale and fixed indentation

Commit:
just return fst_list and comment out print statements

Commit:
temp

Commit:
Merge branch 'as_dev' into ancestral_sample

* as_dev: (29 commits)
  just return fst_list and comment out print statements
  eliminated last reference to burn-in scale and fixed indentation
  (test_evolve and evolve_arg) moved args.anc_sam1/2 manipulation back to test_evolve, add burn_in generations to the args.anc_sam1/2 (argevolver) for max sample index add pop_size1 and 2 instead of just using __pop.N
  syntax and bug fixes
  first commit for multi-sample fst analysis
  change results to evolver
  create sample list from node list
  fix sample dtype
  fix sample taking
  moved anc_sam conversion to int to evolve_arg and fixed indexing in sample_index
  fixing parsed args to args
  fixing tabs
  made sure anc_sampling was of type int
  test evolver in interactive window
  name error
  absolutely should fix dtype error
  should fix dtype error
  hopefully a fix to the numpy dtype error
  fix seeds and add print
  putting back in accidently removed lines
  ...

Commit:
added concurrent futures

Commit:
fixes

Commit:
define fst_list

Commit:
Merge branch 'as_dev' into ancestral_sample

Commit:
print mean, median, and std fst results out to file

Commit:
allow setting of initial migration rate (good for when I wan the sister populations to have the same size)

Commit:
better migration checking when pop 2 doesn't exist

Commit:
output linearized fst results and expected linearized fst results for each sample according to their population and sample generation

Commit:
single population simulation

Commit:
changes to test comparison with coalescent theory for sister populations:

adding a mode whereby at the creation of population 2 if m12 is 1, then the ancestral population essentially duplicates so that, in a flat demography, there is never any reduction in effective population size for either population

Commit:
adding josh schraiber's continuity to project

Commit:
adding simulation command lines

Commit:
initial stab at full integration of continuity with project, probably needs a lot of cleanup and corrections

Commit:
small fixes and testing lines (reads not being sliced right)

Commit:
further attempts to get Josh's continuity program working ... still doesn't

ghost admixture had to comment out the time_out_errors.write function call as pickle couldn't pickle it when sending it out for multiprocessing

still ends up giving an error:

"""
Traceback (most recent call last):
  File "/home/dlawrie/miniconda3/envs/fwdpy11_arg/lib/python3.6/site-packages/joblib/externals/loky/process_executor.py", line 421, in _process_worker
    r = call_item.fn(*call_item.args, **call_item.kwargs)
  File "/home/dlawrie/miniconda3/envs/fwdpy11_arg/lib/python3.6/site-packages/joblib/_parallel_backends.py", line 563, in __call__
    return self.func(*args, **kwargs)
  File "/home/dlawrie/miniconda3/envs/fwdpy11_arg/lib/python3.6/site-packages/joblib/parallel.py", line 261, in __call__
    for func, args, kwargs in self.items]
  File "/home/dlawrie/miniconda3/envs/fwdpy11_arg/lib/python3.6/site-packages/joblib/parallel.py", line 261, in <listcomp>
    for func, args, kwargs in self.items]
  File "run_ghost_admixture_continuity_job.py", line 28, in sim_ghost_admixture
    params_pop_sim_free = optimize_pop_params_error_parallel(freqs_sim,read_list_sim,num_core=1,detail=0,continuity=False)
  File "/home/dlawrie/Dropbox/fwdpy11_arg_example/ancient_genotypes.py", line 419, in optimize_pop_params_error_parallel
    min_a, max_a, min_d, max_d = get_bounds_reads(read_lists)
  File "/home/dlawrie/Dropbox/fwdpy11_arg_example/ancient_genotypes.py", line 317, in get_bounds_reads
    max_a.append(max(map(lambda x: np.amax(x[:,:,0]),reads[i])))
ValueError: max() arg is an empty sequence
"""

test_evolve gives same error when updated to ghost_admixture's  read output and run_continuity fails at get_read_dict

Commit:
created new split/recover and split migration models so that ancestral population can permanently lose individuals after split or recover the population size afterwards

Commit:
allow selection coefficient to be set by user and make use of fwdpy11 regions

Commit:
got separate msprime simulation working (same interface for simplicity)

test_evolve.py: fixed duplicate interface (replicates and iterations), fixed applying burn-in time to right sampling variable

Commit:
fixes

test_msprime: fixing two sample Fst, adding recombination

test_evolve: only throw migration generation error if migration rates are above 1 (and population exists which is checked now in previous if statement)

both: fix expected Fst split times

Commit:
allow number of generations in flat demography to be set by user

Commit:
add option to create custom filename for outfile

Commit:
indentation fix

Commit:
add user defined boolean where split rate is an exact rather than random variable

i.e. the size of the population splitting from population 1 is exactly (split rate)*(pop1 size at time of split) as opposed to being poisson distributed)
this is to make it more comparable to msprime splitting (well technically msprime merging)

Commit:
upgraded msprime simulation to model split_loss and split_recovery models

split_recovery doesn't seem to be working (Fst is lower when it should be higher), could be a result of continuous time versus discrete time problems, a mistake in my coding, or a problem in msprime, unsure

DL42 committed:
fixing population split non-recovery

all non-duplicating split rates are done by expectation calculated from the ratio of the two population sizes

for non-recovery, non-duplicating splits, the population size is permantly reduced by subtracting the number of individuals in the splitting population (popualtion 2) from the ancestral population (population 1)

fixing off-by-one error in the number of generations

DL42 committed:
add tennyson model to msprime

DL42 committed:
simulation commands and results

Commit:
adding option for two-locus simulation

uses regions to simulate two loci with no recombination within each locus but recombination between
first locus consists of selected mutations
second locus consists of neutral mutations
middle region is purely for recombination, no mutations

when generating neutral mutations using msprime, mutation rate is boosted by neutral locus size
as mutations are generated for the whole region between 0 and 1
but then mutations not in neutral locus are filtered out

Commit:
flanking, recombining deleterious regions w/ central, non-recombining neutral region

in arg_evolver replaced parsed_args with args

DL42 committed:
change multiplicative gvalue to 2.0

DL42 committed:
changed name of two_loc param to mul_loc, made region_break exactly 1/3

name change is because the Hudson and Kaplan 95 model the region is supposed simulate is better described as a multilocus model with a single, non-recombining neutral locus in the middle of m deleterious loci, taken to the limit as m -> inf. Thus the final effect is a single, non-recombining neutral locus, flanked by two equal sized, recombining deleterious loci

DL42 committed:
changed -mloc parameter name to avoid conflicts

molpopgen commented on fwdpy11_arg_example/evolve.cc:
```diff
@@ -0,0 +1,168 @@
+#include <fwdpy11/sim_functions.hpp>
+#include <fwdpy11/policies/mutation.hpp>
+#include <fwdpp/poisson_xover.hpp>
+#include <fwdpp/sugar/popgenmut.hpp>
```
Get rid of /sugar here -- that is gone.   It is just `fwdpp/popgenmut.hpp`

molpopgen commented on fwdpy11_arg_example/evolve.cc:
```diff
@@ -0,0 +1,168 @@
+    const auto bound_mmodel = fwdpp::extensions::bind_dmm(rng.get(), mmodel);
+    const auto bound_rmodel = [&rng, &rmodel]() { return rmodel(rng.get()); };
+    
+    const auto ff = fwdpp::multiplicative_diploid(1.0);   
```
This should be `fwdpp::multuplicative_diploid(fwdpp::fitness(1.0))`

Commit:
add cores option, fixes compilation on new fwdpy11 versions

also: notes gv is set on C++ side, notes for cluster usage

DL42 committed:
BGS command lines

also mispelled multiplicative

DL42 committed:
first simulation script

added comment in hpc4david_md

DL42 committed:
fixing BGC->BGS, fixing shell commands

max workers in test_evolve now minimum of ncores possible and number of replicates

Commit:
conversion to msprime 0.6.1

Commit:
new shell scripts

Commit:
check error

Commit:
fixes and mutation printing

Commit:
tried to use table collection API, but no dice

Commit:
deduplicate sites, non-discarding of populations during simplification

other minor fixes and removal of checks

Commit:
removed population_table function, unnecessary, handled in ArgEvolver now

Commit:
updated seed values for shellcmd files and added R values to test

DL42 committed:
results

Commit:
removed the call to make a python interactive after running bash script (forms an error file on cluster)

Commit:
updated shell commands for interleaved model

adds more cores, number replicates determined by using R11 as base r = (std/0.00632455)^2
std is the standard deviation of the experiment under 40 replicates
(if the std of a lower R was lower than a higher R, the higher R's std was used)
the constant 0.006... was calculated as the standard error for R11 0.04/sqrt(40)
this somewhat normalizes the standard error across the interleaved models

Commit:
updated shell commands for HK95 model

adds more cores, number replicates determined by using R11 as base r = (std/0.024375)^2
std is the standard deviation of the experiment under 80 replicates
(if the std of a lower R was lower than a higher R, the higher R's std was used)
the constant 0.02... was calculated as the standard error for R11 0.39/sqrt(256)
maximum number of replicates was 1000
this somewhat normalizes the standard error across the HK models
will be ~4x higher standard error compared to interleaved model

Commit:
moved small replicate results to store them for later if needed

Commit:
decreased number of cores, sometimes 128 ran out of memory

DL42 committed:
large replicate results, missing HKBGS_R2 because I set R to 11000

Commit:
fixes R parameter for HKBGS_R2

Commit:
Results for HKBGS_R2

Commit:
r plot scripts and tables

Commit:
small fix to R script

Commit:
final improvement to R script

Commit:
no pop size change demography model with background selection

Commit:
fixes to command line and shell commands (recombination rate and misc)

Commit:
simulation results no popsize changes

Commit:
print sample pi for every sample

Commit:
fix

Commit:
results and fixes

Commit:
long burn-in HK results

Commit:
delete file

Commit:
upgraded test_msprime to msprime v0.6.1 and pi_array, removed unneeded code from test_evolve.py, created command to run msprime with no recombination

Commit:
fix in test_msprime, changed split rate to 1 (duplication)

split rate:  test_evolve unaffected as it manipulates split_rate to do the right thing, but shellcmd for msprime needs it

Commit:
code to output tree sequences to files

Commit:
tree sequences are not pickle-able, output ts file in run_sim

Commit:
test_msprime fixes and results

Commit:
further fixes to test_msprime and removal of extra zeros from results

Commit:
fixes to ts_output and shell commands

Commit:
got tree sequence output for HK and IBGS R1, rsynced to Irvine/BGS_results\n\nstatistic results equivalent to previous sim runs, not committed

Commit:
continuity analyses successfully run in python 3 for both my simulation and continuity simulation

Commit:
fixes continuity analyses so I can take multiple ancient samples and compare them (one at a time) to the first (modern) sample

comments out print statement in ancient genotypes

Commit:
output continuity results

Commit:
fixes population table when initializing from neutral tree sequence

Commit:
make simulation like mine

Commit:
continuity simulation working!

Commit:
fixing random number seeds in test_evolve so continuity within test_evolve has the right distribution of random numbers (previously was generating too similar random numbers across simulations for continuity within test_evolve)

Commit:
fixed msprime's current population sampling to be the same as fwdpy11's (haploid) ... doesn't seem to change Fst results at all

Commit:
outputting branch length estimations for split times in addition to likelihoods

eliminated unused program

Commit:
preparing for cluster (no joblib/matplotlib), cluster commands

Commit:
fixing cluster commands for fwdpy11 and num_process/replicated for continuity

Commit:
stop openBlas from spamming threads in numpy calculations

Commit:
continuity results for continuity and fwdpy11 with different rates of recombination

Commit:
increase replicates to 10,0000

Commit:
trying to get around msprime simulation error where a small number of simulations fail because the parent gets a time < than a child node

Commit:
forgot that sys.argv[0] is the file name

Commit:
large replicate sim results, fix to continuity shell commands

Commit:
deleted excess simulation functions, added test for bug in msprime when file is run by python as main

Commit:
large replicate sim results for continuity

Commit:
Merge branch 'ancestral_sample' of https://github.com/DL42/fwdpy11_arg_example into ancestral_sample

Commit:
combined continuity msprime sim results files

Commit:
BGS continuity tests for demographic model 2C with recombination rate 11000

Commit:
msprime doesn't like seeds of 0

Commit:
BGS continuity results

Commit:
reorganize files, new R figure code, fix mistake in test_msprime, fix 2E

msprime: for split populations, mutation rate was being set by final population 1 pop size instead of initial population pop 1 size, only affects Figure 2B,2D,2E
2E: simulations was being called with split-recovery model (which doesn't currently exist in fwdpy11 as of right now), should've been called with pure split model

Commit:
reorganize HKBGS results for BGS results

Commit:
rename of folder

Commit:
continuity BGS with duplication demography 1

Commit:
new msprime results

Commit:
continuity BGS results for demography scenario 1 + nohup_msprime_2B

Commit:
corrected sim2C to sim1

Commit:
forgot to add to previous commit

Commit:
2C > 2B

Commit:
continuity BGS with duplication demography 1 and recombination rate 11000

Commit:
eliminated extra file

Commit:
continuity BGS2 rec111 results + fwdpy11 neutral 2E

Commit:
reorganize HK results and add csv results and modify gitignore

Commit:
add and modify rplots

Commit:
rename 2C to 2B

Commit:
made separate axes for graphs

Commit:
multipanel plot 2

Commit:
plot 1 tiff

Commit:
plot 2 reformatted

Commit:
convert to jpg for google docs, regularize sizes, make plot titles "A", "B" bold in neutral_2 plot

Commit:
ignore pptx and tex files, have LaTex output figures using tikzLatex package, use eta for Fst, add symbols for BGS plots, remove msprime results from plot2.pdf, put non-duplication msprime vs fwdpy11 into supplemental figure

Commit:
new combined plot 1

Commit:
increase replicates BGS_1_rec_3 by 16x for 1/4 se

Commit:
new results

Commit:
rename sim2C to sim2B files

Commit:
new continuity BGS plots

Commit:
changing legend keyheight for linetype legend (spacing out neutral and theoretical expectation) and increasing pdf size for some of the plots

Commit:
regularizing BGS plots and legends

Commit:
r_plot for pi
